File: process_sprites.py
import os
from PIL import Image

artifacts_dir = r"C:\Users\home\.gemini\antigravity\brain\98c388c2-3820-487f-9d01-24baf58a2781"
out_dir = r"D:\isaac-godot\assets\sprites"

if not os.path.exists(out_dir):
    os.makedirs(out_dir)

files = {
    "player_rogue": "player.png",
    "projectile_dagger": "dagger.png",
    "enemy_chaser": "skeleton.png",
    "enemy_shooter": "goblin.png",
    "enemy_tank": "orc.png",
    "enemy_flanker": "bat.png",
    "boss_necromancer": "necromancer.png"
}

def remove_white(img_path, out_path):
    img = Image.open(img_path).convert("RGBA")
    data = img.getdata()
    new_data = []
    for item in data:
        # Check if it's very close to white
        if item[0] >= 220 and item[1] >= 220 and item[2] >= 220:
            new_data.append((255, 255, 255, 0))
        else:
            new_data.append(item)
    img.putdata(new_data)
    
    # create cropped bounding box to remove excess transparency
    bbox = img.getbbox()
    if bbox:
        img = img.crop(bbox)
        
    # Standardize size for enemies to roughly 64x64 or aspect ratio
    # Just save raw cropped for now to maintain aspect ratio
    img.save(out_path, "PNG")

for prefix, final_name in files.items():
    latest = None
    latest_time = 0
    for file in os.listdir(artifacts_dir):
        if file.startswith(prefix) and file.endswith(".png"):
            path = os.path.join(artifacts_dir, file)
            t = os.path.getmtime(path)
            if t > latest_time:
                latest_time = t
                latest = path
    if latest:
        remove_white(latest, os.path.join(out_dir, final_name))
        print(f"Processed {final_name}")

File: process_sprites.js

const Jimp = require('jimp');

async function processImage(inputPath, outputPath) {
    try {
        console.log(`Starting: ${inputPath}`);
        const image = await Jimp.read(inputPath);

        // Remove white background
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];

            // If the pixel is very white, make it fully transparent
            if (r > 240 && g > 240 && b > 240) {
                this.bitmap.data[idx + 3] = 0;
            }
        });

        await image.writeAsync(outputPath);
        console.log(`Success processing to: ${outputPath}`);
    } catch (e) {
        console.error(`Error processing ${inputPath}: ${e.message}`);
    }
}

const images = [
    { in: 'C:/Users/home/.gemini/antigravity/brain/5cc6f686-4995-4b32-b6ec-465968c1e545/player_sprite_v2_1772732649696.png', out: 'D:/isaac-godot/assets/player_sprite_new.png' },
    { in: 'C:/Users/home/.gemini/antigravity/brain/5cc6f686-4995-4b32-b6ec-465968c1e545/enemy_basic_v2_1772732672296.png', out: 'D:/isaac-godot/assets/enemy_basic.png' },
    { in: 'C:/Users/home/.gemini/antigravity/brain/5cc6f686-4995-4b32-b6ec-465968c1e545/monstro_sprite_v2_1772732689317.png', out: 'D:/isaac-godot/assets/monstro_sprite.png' }
];

(async () => {
    for (const img of images) {
        await processImage(img.in, img.out);
    }
})();
